### PR TITLE
common/00-libdir: only create lib{32,64} symlink if not exists

### PR DESCRIPTION
**Pull request by sgn**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: common/hooks/pre-install/00-libdir.sh [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # This hook creates the wordsize specific libdir symlink.
 
 hook() {
-	if [ "${pkgname}" != "base-files" ]; then
+	if [ -L ${PKGDESTDIR}/usr/lib${XBPS_TARGET_WORDSIZE} ]; then
+		return 0
+	elif [ "${pkgname}" != "base-files" ]; then
```
Comment (ericonr) [inline common/hooks/pre-install/00-libdir.sh:6]: This should only happen if you're rebuilding in a dirty destdir... Maybe it should have a dotfile that tracks the hook already having been run?

Comment (sgn) [reply common/hooks/pre-install/00-libdir.sh:6]: Yes, but my use case is:
```sh
$ ./xbps-src pkg foo
$ do-some-test-and-hate-the-result
$ fix-some-stuff
$ ./xbps-src install foo
$ inspect $DESTDIR
$ ./xbps-src -f pkg foo # this steps will run into error :-[p

Comment (sgn) [reply common/hooks/pre-install/00-libdir.sh:6]: This is a hack, not a proper fix, I think.

Comment (ericonr) [reply common/hooks/pre-install/00-libdir.sh:6]: Maybe leave a comment about it being a hack? The issue with these hooks is that some of them we'd like to run only once, and others we'd like to run despite it having been run already, so a general `.pre-pkg-hook-*` mechanism isn't enough :/

Still, I thought `ln -sf` avoid issues with creating symlinks inside symlinks.

Comment (sgn) [reply common/hooks/pre-install/00-libdir.sh:6]: Yay, the keyword is _thought_

**Comment by q66** [comment]
what's the point of doing this? it should make no semantical difference (the symlink is made with `ln -sf`)

**Comment by sgn** [comment]
On 2021-02-09 01:47:58-0800, Daniel Kolesa <notifications@github.com> wrote:
> what's the point of doing this? it should make no semantical difference (the symlink is made with `ln -sf`)

To prevent the existance of `/usr/lib{32,64}/lib` with

./xbps-src install foo && ./xbps-src -f pkg foo

**Comment by q66** [comment]
i see no way the script can cause that, it never creates `lib`, only `lib32/64`

**Comment by ericonr** [comment]
@q66 you can see it happening by exiting with an error from a `do_install` and then running it again.

**Comment by sgn** [comment]
On 2021-02-10 03:19:24-0800, Daniel Kolesa <notifications@github.com> wrote:
> i see no way the script can cause that, it never creates `lib`, only `lib32/64`

$ mkdir -p /tmp/lib
$ ln -sf lib /tmp/lib64
$ ln -sf lib /tmp/lib64

**Comment by q66** [comment]
ah, i see, yeah

**Pull request merged**